src/Workspaces/Core/Portable/TemporaryStorage/TemporaryStorageServiceFactory.cs

```diff
@@ -206,12 +209,20 @@ public TemporaryTextStorage(TemporaryStorageService service, string storageName,
 
                 public ImmutableArray<byte> GetChecksum()
                 {
-                    if (_checksum.IsDefault)
+                    if (_checksumData.IsDefault)
                     {
-                        ImmutableInterlocked.InterlockedInitialize(ref _checksum, ReadText(CancellationToken.None).GetChecksum());
+                        if (_checksum is not null)
+                        {
+                            Debug.Assert(_checksum.ToImmutableArray().SequenceEqual(ReadText(CancellationToken.None).GetChecksum()));
```

Comment:
so i really like this assert.  i'm curious if it fired in our test suite.  sounds like since we had a bug it should have.  if it didn't fire, does tha mean we don't jhave enough coverage?

i wish we could turn on something like this in release, but i can't think how without tanking perf, which is the entire purpose of hydrating with this value in the first place.

Reply:
Yes, the original test failures were triggered by this assertion.

Reply:
perfect.